update-common/src/artifacts/update_plan.rs

```diff
@@ -2359,6 +2486,67 @@ mod tests {
         logctx.cleanup_successful();
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_split_control_plane() {
+        const VERSION_0: SemverVersion = SemverVersion::new(0, 0, 0);
+        const ZONES: &[(&str, &[u8])] =
+            &[("first", b"the first zone"), ("second", b"the second zone")];
+
+        let logctx = test_setup_log("test_split_control_plane");
+
+        let mut cp_builder = CompositeControlPlaneArchiveBuilder::new(
+            Vec::new(),
+            MtimeSource::Now,
+        )
+        .unwrap();
+        for (name, data) in ZONES {
+            cp_builder
+                .append_zone(
+                    name,
+                    CompositeEntry { data, mtime_source: MtimeSource::Now },
```

Comment:
The unit test failure seems legit - does `data` need to be a tarball containing an `oxide.json`?